vaadin-grid-flow-parent/vaadin-grid-flow-integration-tests/src/test/java/com/vaadin/flow/component/treegrid/it/TreeGridScrollToIT.java

```diff
@@ -0,0 +1,158 @@
+
+    private TestBenchElement scrollToStartButton;
+
+    private TestBenchElement scrollToEndButton;
+
+    private TestBenchElement scrollToIndexInput;
+
+    @Before
+    public void init() {
+        open();
+        waitUntil(e -> $(TreeGridElement.class).exists(), 500);
+        grid = $(TreeGridElement.class).first();
+        expandAllButton = $("button").id("expand-all");
+        scrollToStartButton = $("button").id("scroll-to-start");
+        scrollToEndButton = $("button").id("scroll-to-end");
+        scrollToIndexInput = $("input").id("scroll-to-index");
+
+    }
+
+    @Test
+    public void expandAll_scrollToEnd_correctLastVisibleItem()
+            throws InterruptedException {
+        expandAllButton.click();
+
+        scrollToEndButton.click();
+        Thread.sleep(2000);
+
+        Assert.assertEquals("Son 49/19/19",
+                getCellContent(grid.getLastVisibleRowIndex()));
+    }
+
+    @Test
+    public void scrollToEnd_correctLastVisibleItem() {
+        scrollToEndButton.click();
+
+        Assert.assertEquals("Granddad 49",
+                getCellContent(grid.getLastVisibleRowIndex()));
+    }
+
+    @Test
+    public void scrollToEnd_scrollToStart_correctFirstVisibleItem()
+            throws InterruptedException {
+        scrollToEndButton.click();
+        Thread.sleep(200);
+
+        scrollToStartButton.click();
+
+        Assert.assertEquals("Granddad 0",
+                getCellContent(grid.getFirstVisibleRowIndex()));
+    }
+
+    @Test
+    public void expandAll_scrollToIndex30_correctFirstVisibleItem()
+            throws InterruptedException {
+        expandAllButton.click();
+
+        scrollToIndexInput.sendKeys("30", Keys.TAB);
+        Thread.sleep(2000);
+
+        Assert.assertEquals("Granddad 30",
+                getCellContent(grid.getFirstVisibleRowIndex()));
+    }
+
+    @Test
+    public void scrollToIndex30_correctFirstVisibleItem() {
+        scrollToIndexInput.sendKeys("30", Keys.TAB);
+
+        Assert.assertEquals("Granddad 30",
+                getCellContent(grid.getFirstVisibleRowIndex()));
+    }
+
+    @Test
+    public void expandAll_scrollToIndex30_1_correctFirstVisibleItem()
+            throws InterruptedException {
+        expandAllButton.click();
+
+        scrollToIndexInput.sendKeys("30-1", Keys.TAB);
+        Thread.sleep(2000);
+
+        Assert.assertEquals("Dad 30/1",
+                getCellContent(grid.getFirstVisibleRowIndex()));
+    }
+
+    @Test
+    public void scrollToIndex30_1_correctFirstVisibleItem() {
+        scrollToIndexInput.sendKeys("30-1", Keys.TAB);
+
+        Assert.assertEquals("Granddad 30",
+                getCellContent(grid.getFirstVisibleRowIndex()));
+    }
+
+    @Test
+    public void expandAll_scrollToIndex30_1_1_correctFirstVisibleItem()
+            throws InterruptedException {
+        expandAllButton.click();
+
+        scrollToIndexInput.sendKeys("30-1-1", Keys.TAB);
+        Thread.sleep(2000);
```

Comment:
Would `waitUntil(...)` work here instead of a constant timeout?

Reply:
Sure, I updated the PR. Note that the extra `grid.hasRow(...)` call is there to make sure the row is ready and avoid failure due to exceptions.